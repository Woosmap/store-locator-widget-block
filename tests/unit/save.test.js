import {render, cleanup} from "@testing-library/react";
import StoreLocatorBlockSave from '../../src/save';

describe("StoreLocatorBlockSave component", () => {
	afterEach(cleanup);

	it("matches snapshot with default attributes", () => {
		const attributes = {
			height: 300,
			woosmapView: {
				initialCenter: {lat: 50, lng: 0},
				initialZoom: 13,
			},
			theme: {
				primary_color: '#ffffff'
			},
			apiKey: ''
		};

		const {asFragment} = render(
			<StoreLocatorBlockSave
				{...{
					attributes,
					clientId: "random-id",
					className: "wp-blocks-whatever",
				}}
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});

	it("matches snapshot with provided attributes", () => {
		const attributes = {
			height: 400,
			woosmapView: {
				initialCenter: {lat: 51.5074, lng:  0.1278},
				initialZoom: 15,
			},
			theme: {
				primary_color: '#ffffff'
			},
			apiKey: 'your-api-key'
		};

		const {asFragment} = render(
			<StoreLocatorBlockSave
				{...{
					attributes,
					clientId: "random-id",
					className: "wp-blocks-whatever",
				}}
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
