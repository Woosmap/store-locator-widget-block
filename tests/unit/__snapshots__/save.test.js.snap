// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocatorBlockSave component matches snapshot with default attributes 1`] = `
<DocumentFragment>
  <div
    data-api-key=""
    data-theme="{"primary_color":"#ffffff"}"
    data-woosmapView="{"initialCenter":{"lat":50,"lng":0},"initialZoom":13}"
    id="storeLocatorWidget"
    style="height: 300px;"
  />
</DocumentFragment>
`;

exports[`StoreLocatorBlockSave component matches snapshot with provided attributes 1`] = `
<DocumentFragment>
  <div
    data-api-key="your-api-key"
    data-theme="{"primary_color":"#ffffff"}"
    data-woosmapView="{"initialCenter":{"lat":51.5074,"lng":0.1278},"initialZoom":15}"
    id="storeLocatorWidget"
    style="height: 400px;"
  />
</DocumentFragment>
`;
